x-pack/plugins/index_lifecycle_management/public/url_generator.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'kibana/public';
+import { UrlGeneratorsDefinition } from '../../../../src/plugins/share/public/';
+import {
+  getPoliciesListPath,
+  getPolicyCreatePath,
+  getPolicyEditPath,
+} from './application/services/navigation';
+import { MANAGEMENT_APP_ID } from '../../../../src/plugins/management/public';
+import { SetupDependencies } from './types';
+import { PLUGIN } from '../common/constants';
+
+export const ILM_URL_GENERATOR_ID = 'ILM_URL_GENERATOR_ID';
+
+export enum ILM_PAGES {
+  LIST = 'policies_list',
+  EDIT = 'policy_edit',
+  CREATE = 'policy_create',
+}
+
+export interface IlmPoliciesListUrlGeneratorState {
+  page: ILM_PAGES.LIST;
+  absolute?: boolean;
+}
+
+export interface IlmPolicyEditUrlGeneratorState {
+  page: ILM_PAGES.EDIT;
+  policyName: string;
+  absolute?: boolean;
+}
+
+export interface IlmPolicyCreateUrlGeneratorState {
+  page: ILM_PAGES.CREATE;
+  absolute?: boolean;
+}
+
+export type IlmUrlGeneratorState =
+  | IlmPoliciesListUrlGeneratorState
+  | IlmPolicyEditUrlGeneratorState
+  | IlmPolicyCreateUrlGeneratorState;
+
+export class IlmUrlGenerator implements UrlGeneratorsDefinition<typeof ILM_URL_GENERATOR_ID> {
```

Comment:
I probably would have used a factory pattern here to capture `getAppBasePath`.

```ts
const createIlmUrlGenerator = (getApppBasePath: getAppBasePath: (absolute?: boolean) => Promise<string>): UrlGeneratorsDefinition<typeof ILM_URL_GENERATOR_ID> => {
   return { id: ILM_URL_GENERATOR_ID, /* etc */ }
}
```

I think it would result in slightly less code. But I believe this is mostly stylistic preference!